src/blocknative.rs

```diff
@@ -132,14 +132,21 @@ impl BlockNative {
 
 #[async_trait::async_trait]
 impl GasPriceEstimating for BlockNative {
-    async fn estimate_with_limits(&self, _gas_limit: f64, time_limit: Duration) -> Result<f64> {
+    async fn estimate_with_limits(
+        &self,
+        _gas_limit: f64,
+        time_limit: Duration,
+    ) -> Result<(f64, f64, f64)> {
```

Comment:
In my experience, in a triple of identical types it is very easy to confuse the caller what value is what. Let's maybe use a named struct instead? Also, if the second two values are 0 in case of "no EIP 1559 support", maybe they should be wrapped in an Option to make that more explicit?

Reply:
Yes, I was not sure if custom structures are appropriate to send across this crate. Fixed.